lib/ansible/module_utils/facts/timeout.py

```diff
@@ -50,6 +50,8 @@ def wrapper(*args, **kwargs):
             except multiprocessing.TimeoutError:
                 # This is an ansible.module_utils.common.facts.timeout.TimeoutError
                 raise TimeoutError('Timer expired after %s seconds' % timeout_value)
+            finally:
+                pool.terminate()
```

Comment:
might want to use try/except AttributeError pass as the pool does not always have '.terminate' available (we had to do this in executor).

Reply:
This should be fine.  In executor, we're using ```worker.terminate()```.  Here we're using ```pool.terminate()```.  ```pool.terminate()``` should always exist.  From reading the multiprocessing code it looks like ```worker.terminate()``` only exists for processes.  Not for threads.

In reading up about this issue and also the thread code, though, I wonder if terminate() might be the source of the problems you had in the parallel mount PR.  Threads in Python don't seem to have a clean way to terminate so if the code somehow causes a join() to be called, we'd then block on the thread exiting.  I'll run the terminate code through me test case before merging.  (Also, from my reading of the code, pool.terminate() might not guarantee thread termination...)  I'll experiment more before merging.

Reply:
For this case, signals might still be the answer, since its a single thread there isn't contention on which part of the code receives it, unlike mine which uses multiple threads.